src/fourc_webviewer/fourc_webserver.py

```diff
@@ -228,61 +227,149 @@ def sync_server_vars_from_state(self):
+                )  # in PolyData
+                self._actors["dc_geometry_entities"][
+                    (dc_entity["entity"], dc_entity["geometry_type"])
+                ] = self._server_vars["render_window"].add_mesh(
+                    glyphs,
+                    color="navy",
+                    opacity=1.0,
+                    render=False,
+                )
 
-        # get coords of node with prescribed result description
-        self._server_vars["pv_selected_result_description_node_coords"] = (
-            self._server_vars["pv_mesh"].points[
-                self.state.result_description_section[
-                    self.state.selected_result_description_id
-                ]["PARAMETERS"]["NODE"]
-                - 1,
+        self._actors["result_description_nodes"] = {}
+        all_result_descriptions = self.state.result_description_section.keys()
+
+        for dc in all_result_descriptions:
+            node_coords = problem_mesh.points[
+                self.state.result_description_section[dc]["PARAMETERS"]["NODE"] - 1,
                 :,
             ]
-        )
+            self._actors["result_description_nodes"][dc] = self._server_vars[
+                "render_window"
+            ].add_mesh(
+                pv.Sphere(
+                    center=node_coords,
+                    radius=pv_render.get_problem_length_scale(problem_mesh)
+                    * pv_render.PV_SPHERE_FRAC_SCALE,
+                ),
+                color="deepskyblue",
+                render=False,
+            )
+        self.update_pyvista_render_objects()
 
-        # update plotter / rendering
-        pv_render.update_pv_plotter(
-            self._server_vars["render_window"],
-            self._server_vars["pv_mesh"],
-            self._server_vars["pv_selected_material_mesh"],
-            self._server_vars["pv_selected_dc_geometry_entity"],
-            self._server_vars["pv_selected_result_description_node_coords"],
-        )
+    def update_pyvista_render_objects(self):
+        """Update/ initialize pyvista view objects (reader, thresholds, global
+        COS, ...) for the rendered window. The saved vtu file path is hereby
+        utilized.
+
+        Args:
+            init_rendering (bool): perform initialization tasks? (True:
+            yes(also when loading a new file) | False: no -> only updating)
+        """
+        # 70ms  for tutorial_solid.4C.yaml DESIGN CONDITIONS/SURF/E2    on LasseLaptop
+        # 0ms   for tutorial_solid.4C.yaml PROBLEM TYPE                 on LasseLaptop
+        legend_items = []
+
+        for dc in self._actors["result_description_nodes"].values():
+            dc.SetVisibility(False)
+        if (
+            self.state.selected_result_description_id
+            in self._actors["result_description_nodes"].keys()
+            and self.state.selected_result_description_id
+            and self.state.selected_main_section_name == "RESULT DESCRIPTION"
```

Comment:
Shouldn't the checks be reversed (at least the first two ones, but also the third), i.e.,:

```suggestion
           self.state.selected_main_section_name == "RESULT DESCRIPTION and
            self.state.selected_result_description_id
            and self.state.selected_result_description_id
            in self._actors["result_description_nodes"].keys()"
```

Reply:
Sure, if you think it reads better. Logically its the same of course.